### PR TITLE
Release 1.25

### PR DESCRIPTION
**Pull request by sethmlarson**
Testing our new release system! I figure we should wait for Monday to release this as it's quite a large one. cc @shazow @theacodes 

Requests already has a PR for unpinning: https://github.com/kennethreitz/requests/pull/5063

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -1,34 +1,32 @@
 Changes
 =======
 
-dev (master)
```
Comment (shazow) [inline CHANGES.rst]: Is the dev section going away in master? Or was this meant to be merged into release?

Comment (sethmlarson) [reply CHANGES.rst]: Per #1392 we won't be tagging releases on the release branch anymore, will be doing so on the master branch. Whenever we do additional development on master we'll have to add the `dev (master)` section back to whichever changelog entry comes next.

Comment (shazow) [reply CHANGES.rst]: Ah, didn't realize we'd be getting rid of `__version__ = 'dev'` in master. Historically, it has been very useful to know if the install version is directly from the repo vs a tagged release. Will the immediate next release also bump the `__version__` to an unreleased version too?

Don't think I mentioned that in #1392 but I won't stand in the way. :)

**Comment by theacodes** [comment]
We can send a follow-up PR right after this to restore `__version__` to `dev` and restore the `dev` entry in `CHANGELOG`.

**Pull request merged**